tests/sys_crypto/tests-crypto-modes-ccms.c

```diff
@@ -0,0 +1,318 @@
+static const uint8_t TEST_IEEE_0537R2_1_EXPECTED[] = {
+    /* AAD */
+    0x08, 0xD0, 0x84, 0x21, 0x43, 0x01, 0x00, 0x00,
+    0x00, 0x00, 0x48, 0xDE, 0xAC, 0x02, 0x05, 0x00,
+    0x00, 0x00, 0x55, 0xCF, 0x00, 0x00, 0x51, 0x52,
+    0x53, 0x54,
+    /* MAC */
+    0x22, 0x3B, 0xC1, 0xEC, 0x84, 0x1A, 0xB5, 0x53,
+};
+static const size_t TEST_IEEE_0537R2_1_EXPECTED_LEN = 34;
+
+/* IEEE 0537r2 */
+static const uint8_t TEST_IEEE_0537R2_2_KEY[] = {
+     0xC0, 0xC1, 0xC2, 0xC3, 0xC4, 0xC5, 0xC6, 0xC7,
+     0xC8, 0xC9, 0xCA, 0xCB, 0xCC, 0xCD, 0xCE, 0xCF
+};
+static const uint8_t TEST_IEEE_0537R2_2_KEY_LEN = 16;
+static const uint8_t TEST_IEEE_0537R2_2_NONCE[] = {
+    0xAC, 0xDE, 0x48, 0x00, 0x00, 0x00, 0x00, 0x01,
+    0x00, 0x00, 0x00, 0x05, 0x04
+};
+static const uint8_t TEST_IEEE_0537R2_2_NONCE_LEN = 13;
+static const uint8_t TEST_IEEE_0537R2_2_MAC_LEN = 0;
+static const uint8_t TEST_IEEE_0537R2_2_INPUT[] = {
+    /* AAD */
+    0x69, 0xDC, 0x84, 0x21, 0x43, 0x02, 0x00, 0x00,
+    0x00, 0x00, 0x48, 0xDE, 0xAC, 0x01, 0x00, 0x00,
+    0x00, 0x00, 0x48, 0xDE, 0xAC, 0x04, 0x05, 0x00,
+    0x00, 0x00,
+    /* PLAINTEXT */
+    0x61, 0x62, 0x63, 0x64
+};
+static const size_t TEST_IEEE_0537R2_2_INPUT_LEN = 4;
+static const size_t TEST_IEEE_0537R2_2_ADATA_LEN = 26;
+static const uint8_t TEST_IEEE_0537R2_2_EXPECTED[] = {
+    /* AAD */
+    0x69, 0xDC, 0x84, 0x21, 0x43, 0x02, 0x00, 0x00,
+    0x00, 0x00, 0x48, 0xDE, 0xAC, 0x01, 0x00, 0x00,
+    0x00, 0x00, 0x48, 0xDE, 0xAC, 0x04, 0x05, 0x00,
+    0x00, 0x00,
+    /* CIPHERTEXT */
+    0xD4, 0x3E, 0x02, 0x2B,
+};
+static const size_t TEST_IEEE_0537R2_2_EXPECTED_LEN = 30;
+
+/* IEEE 0537r2 */
+static const uint8_t TEST_IEEE_0537R2_3_KEY[] = {
+     0xC0, 0xC1, 0xC2, 0xC3, 0xC4, 0xC5, 0xC6, 0xC7,
+     0xC8, 0xC9, 0xCA, 0xCB, 0xCC, 0xCD, 0xCE, 0xCF
+};
+static const uint8_t TEST_IEEE_0537R2_3_KEY_LEN = 16;
+static const uint8_t TEST_IEEE_0537R2_3_NONCE[] = {
+    0xAC, 0xDE, 0x48, 0x00, 0x00, 0x00, 0x00, 0x01,
+    0x00, 0x00, 0x00, 0x05, 0x06
+};
+static const uint8_t TEST_IEEE_0537R2_3_NONCE_LEN = 13;
+static const uint8_t TEST_IEEE_0537R2_3_MAC_LEN = 8;
+static const uint8_t TEST_IEEE_0537R2_3_INPUT[] = {
+    /* AAD */
+    0x2B, 0xDC, 0x84, 0x21, 0x43, 0x02, 0x00, 0x00,
+    0x00, 0x00, 0x48, 0xDE, 0xAC, 0xFF, 0xFF, 0x01,
+    0x00, 0x00, 0x00, 0x00, 0x48, 0xDE, 0xAC, 0x06,
+    0x05, 0x00, 0x00, 0x00, 0x01,
+    /* PLAINTEXT */
+    0xCE
+};
+static const size_t TEST_IEEE_0537R2_3_INPUT_LEN = 1;
+static const size_t TEST_IEEE_0537R2_3_ADATA_LEN = 29;
+static const uint8_t TEST_IEEE_0537R2_3_EXPECTED[] = {
+    /* AAD */
+    0x2B, 0xDC, 0x84, 0x21, 0x43, 0x02, 0x00, 0x00,
+    0x00, 0x00, 0x48, 0xDE, 0xAC, 0xFF, 0xFF, 0x01,
+    0x00, 0x00, 0x00, 0x00, 0x48, 0xDE, 0xAC, 0x06,
+    0x05, 0x00, 0x00, 0x00, 0x01,
+    /* CIPHERTEXT */
+    0xD8,
+    /* MAC */
+    0x4F, 0xDE, 0x52, 0x90, 0x61, 0xF9, 0xC6, 0xF1
+};
+static const size_t TEST_IEEE_0537R2_3_EXPECTED_LEN = 38;
```

Comment:
Could you maybe add some more test vectors with longer data. Since the CCM implementation had problems in the past in that area, it would be helpful.

Reply:
Does it make sense to add those? They both share the same base code, and those extended length cases are already tested in `tests-crypto-modes-ccms.c`, I could still generate some if you feel its needed.

Reply:
Why not just add one long sequence... Doesn't the above referenced document provide one?

Reply:
Nope, I added all the vectors provided in the document. I added in the last fixup some longer test vectors generated with PyCryptodome.